packages/web-components/fast-foundation/src/tooltip/tooltip.ts

```diff
@@ -0,0 +1,469 @@
+                this.region !== null &&
+                this.region !== undefined &&
+                this.tooltipVisible
+            ) {
+                this.region.anchorElement = this.anchorElement;
+            }
+
+            this.updateLayout();
+        }
+    }
+
+    /**
+     * The current viewport element instance
+     *
+     * @internal
+     */
+    @observable
+    public viewportElement: HTMLElement | null = null;
+    private viewportElementChanged(): void {
+        if (this.region !== null && this.region !== undefined) {
+            this.region.viewportElement = this.viewportElement;
+        }
+        this.updateLayout();
+    }
+
+    /**
+     * @internal
+     */
+    @observable
+    public verticalPositioningMode: AxisPositioningMode = "dynamic";
+
+    /**
+     * @internal
+     */
+    @observable
+    public horizontalPositioningMode: AxisPositioningMode = "dynamic";
+
+    /**
+     * @internal
+     */
+    @observable
+    public horizontalInset: string = "true";
+
+    /**
+     * @internal
+     */
+    @observable
+    public verticalInset: string = "false";
+
+    /**
+     * @internal
+     */
+    @observable
+    public horizontalScaling: AxisScalingMode = "anchor";
+
+    /**
+     * @internal
+     */
+    @observable
+    public verticalScaling: AxisScalingMode = "content";
+
+    /**
+     * @internal
+     */
+    @observable
+    public verticalDefaultPosition: string | undefined = undefined;
+
+    /**
+     * @internal
+     */
+    @observable
+    public horizontalDefaultPosition: string | undefined = undefined;
+
+    /**
+     * @internal
+     */
+    @observable
+    public tooltipVisible: boolean = false;
+
+    /**
+     * Track current direction to pass to the anchored region
+     * updated when tooltip is shown
+     *
+     * @internal
+     */
+    @observable
+    public currentDirection: Direction = Direction.ltr;
+
+    /**
+     * reference to the anchored region
+     *
+     * @internal
+     */
+    public region: AnchoredRegion;
+
+    /**
+     * The timer that tracks delay time before the tooltip is shown on hover
+     */
+    private delayTimer: number | null = null;
+
+    /**
+     * Indicates whether the anchor is currently being hovered
+     */
+    private isAnchorHovered: boolean = false;
+
+    constructor() {
+        super();
+    }
+
+    public connectedCallback(): void {
+        super.connectedCallback();
+        this.anchorElement = this.getAnchor();
+
+        this.updateLayout();
+        this.updateTooltipVisibility();
+    }
+
+    public disconnectedCallback(): void {
+        this.hideTooltip();
+        this.clearDelayTimer();
+        super.disconnectedCallback();
+    }
+
+    /**
+     * invoked when the anchored region's position relative to the anchor changes
+     *
+     * @internal
+     */
+    public handlePositionChange = (ev: Event): void => {
+        this.classList.toggle("top", this.region.verticalPosition === "top");
```

Comment:
Is there a reason this work isn't being done in the template where it is batched?

Reply:
I like that this approach for core functionality does not depend on the template side implementation which can be swapped out.  Should revisit if we suspect a perf cost though.